docs/faq/ImmutableData.md

```diff
@@ -1,245 +1,145 @@
-<a id="is-immutable-js-worth-effort"></a>
-## Is Using Immutable.JS worth the effort?
 
-Yes. Do not underestimate the difficulty of trying to track down a property of your state tree that has been inadvertently mutated. Components will both re-render when they shouldn’t, and refuse to render when they should, and tracking down the bug causing the rendering issue is hard, as the component rendering incorrectly is not necessarily the one whose properties are being accidentally mutated.
+```
+function mutateObj(obj) {
+	obj.key = 'newValue';
+	return obj;
+}
 
-This problem is caused predominantly by bugs in a Redux reducer, which requires immutability. If you mutate your state in a reducer, it can affect a completely different part of your app in seemingly arbitrary ways.
+const param = { key: 'originalValue' };
+const returnVal = mutateObj(param);
 
-With Immutable.JS, this problem simply does not exist, thereby removing a whole class of bugs from your app. This, together with its performance and rich API for data manipulation, is why Immutable.JS is worth the effort.
-<a id="immutable-js-best-practices"></a>
-## What are the Recommended Best Practices for using Immutable.JS with Redux?
+param === returnVal;
+//> true
+```
 
-Immutable.JS can provide significant reliability and performance improvements to your app, but it must be used correctly. Follow these best practices, and you’ll be able to get the most out of it, without tripping up on any of the issues it can potentially cause.
+The shallow check of `param` and `returnValue` simply checks whether both variables reference the same object, which they do.`mutateObj()` may return a mutated version of `obj`, but it's still the same object as that passed in. The fact that its values have been changed within `mutateObj` matters not at all to a shallow check.
 
-#### Never mix plain JavaScript objects with Immutable.JS  
+### How does shallow equality checking with a mutable object cause problems with Redux?
+Redux holds a reference to the `state` object that’s passed into a reducer function, and compares it with the reference that's returned.
 
-Never let a plain JavaScript object contain Immutable.JS properties. Equally, never let an Immutable.JS object contain a plain JavaScript object.
+Thus, if a reducer mutates any of the properties of the `state` object that's passed in, and then returns a reference to the same _newly mutated_ `state` object, Redux's shallow check of the two references will show them as being equal (although the _values_ within the `state` object have changed, the object reference itself has not - it's still the same `state` object, just with different values.) 
 
-#### Make your entire Redux state tree an Immutable.js object
+_In this case, Redux will not be aware of any changes in your reducer, and the store will not update._
```

Comment:
Per prior comment, I don't think this statement is correct.  Mutations are reflected in the resulting store state, because the old objects were mutated.